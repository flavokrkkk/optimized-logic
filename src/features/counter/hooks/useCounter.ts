import { CounterContext } from "@/entities";
import { useContext } from "react";
export const useCounter = () => {
  const context = useContext(CounterContext);

  if (!context) {
    throw new Error("useCounter must be used within a CounterProvider");
  }

  return context;
};
