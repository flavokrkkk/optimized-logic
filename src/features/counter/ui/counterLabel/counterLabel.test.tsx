import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import { CounterProvider } from "@/entities";
import CounterLabel from "../counterLabel/counterLabel";
import { useCounter } from "../../hooks/useCounter";

jest.mock("../../hooks/useCounter", () => ({
  useCounter: jest.fn(),
}));

describe("CounterLabel", () => {
  it("should render the Counter button correctly", () => {
    (useCounter as jest.Mock).mockReturnValue({
      counter: 5,
    });

    render(
      <CounterProvider>
        <CounterLabel />
      </CounterProvider>
    );

    expect(screen.getByText("Counter")).toBeInTheDocument();
  });

  it("should pass the counter value to the Button component", () => {
    const mockCounter = 10;
    (useCounter as jest.Mock).mockReturnValue({
      counter: mockCounter,
    });

    render(
      <CounterProvider>
        <CounterLabel />
      </CounterProvider>
    );
  });

  it("should match snapshot", () => {
    (useCounter as jest.Mock).mockReturnValue({
      counter: 0,
    });

    const { asFragment } = render(
      <CounterProvider>
        <CounterLabel />
      </CounterProvider>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
