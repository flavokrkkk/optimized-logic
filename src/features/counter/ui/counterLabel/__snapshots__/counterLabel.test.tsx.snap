// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CounterLabel should match snapshot 1`] = `
<DocumentFragment>
  <button
    class="btn primary medium enabled"
  >
    <span
      class="content-group"
    >
      Counter
    </span>
    <div
      class="counter primary size-16"
    >
      <span
        class="counter-quantity"
      >
        0
      </span>
    </div>
  </button>
</DocumentFragment>
`;
